gnucash/import-export/import-pending-matches.cpp

```diff
@@ -70,28 +67,40 @@ gnc_import_PendingMatches_remove_match (GNCImportPendingMatches& map,
 {
     g_return_if_fail (match_info);
 
-    auto& pending_matches = gnc_import_PendingMatches_get_value (map, match_info);
+    auto match_guid = gnc_import_PendingMatches_get_key (match_info);
+
+    auto iter = map.find (match_guid);
+    if (iter == map.end())
+        throw std::out_of_range("coding error. trying to remove missing match");
```

Comment:
You can't throw in a function that's called from C, it crashes the program. If you really want to bail out of the import then create an `abort_import` function and call that from here, but that function needs to pop a dialog explaining to the user what went wrong and what to do about it. I suspect even that's a little extreme and that `g_return_if_fail(iter != map.end());` is sufficient.